segmentation/utils/spatial_analysis_utils.py

```diff
@@ -253,3 +253,48 @@ def calculate_enrichment_stats(close_num, close_num_rand):
     dims = ["stats", "rows", "cols"]
     stats_xr = xr.DataArray(stats_data, coords=coords, dims=dims)
     return stats_xr
+
+
+def compute_neighbor_counts(fov_data, dist_matrix, distlim, self_neighbor=True,
+                            cell_label_col="cellLabelInImage"):
+    """Calculates the number of neighbor phenotypes for each cell. The cell counts itself as a neighbor.
+
+    Args:
+        fov_data: data for the current fov, including the cell labels, cell phenotypes, and cell phenotype ID
+        dist_matrix: cells x cells matrix with the euclidian
+            distance between centers of corresponding cells
+        distlim: threshold for spatial enrichment distance proximity
+        self_neighbor: If true, cell counts itself as a neighbor in the analysis.
+        cell_label_col: Column name with the cell labels
+    Returns:
+        counts: data array with phenotype counts per cell
+        freqs: data array with phenotype frequencies of
+            counts per phenotype/total phenotypes for each cell
+        cell_count: current cell in analysis"""
+
+    # TODO remove non-cell2cell lines after our own inputs for functions are created
+    # remove non-cell2cell distances
```

Comment:
I think this comment should be updated because as @ngreenwald and @ackagel discussed below, it's not entirely clear what non-cell2cell distances are being removed and how fov_data[cell_label_col] - 1 specifically indexes rows and columns that are cell2cell distances. This is especially the case when you look at calc_dist_matrix in spatial_analysis_utils where dist_matrix is created. It doesn't seem like that function includes any non-cell2cell distances, at least how I understood it.

In summary, I think it would be helpful to clarify further what non-cell2cell distances means in the context of this function and the function that calls it (create_neighborhood_matrix).

Reply:
Changing the comment to something like 'refine distance matrix to only cover cell ids in fov_data' could be better.  Regardless, I agree that non-cell2cell is pretty non-specific.  idk if that needs to be changed right now, though.  Thoughts @ngreenwald ?

Reply:
I believe what's happening is the following. The distance matrix has the first cell ID with index 0, and doesn't include background as a row. However, the cell IDs start at  1, because background is assigned the value of 0. So we have to subtract 1 to convert from cell_IDs to row indices. 

There are certain Cell_IDs which are filtered out, because they aren't cells. This is an artifact of the previous segmentation which included non-cell objects that were identified by heuristic rules. However, it appears that maybe the distance matrix was created with all objects, not just cell objects? Not 100% sure about this part, but this isn't relevant moving forward because we won't have this issue. However, we're going to wait to remove this until we have our own fleshed out test functions, since right now we want to be able to check against the matlab output